windows-rdp/main.tf

```diff
@@ -0,0 +1,177 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 0.17"
+    }
+  }
+}
+
+variable "agent_id" {
+  type        = string
+  description = "The ID of a Coder agent."
+}
+
+variable "resource_id" {
+  type        = string
+  description = "The ID of the primary Coder resource (e.g. VM)."
+}
+
+variable "admin_username" {
+  type    = string
+  default = "Administrator"
+}
+
+variable "admin_password" {
+  type      = string
+  default   = "coderRDP!"
+  sensitive = true
+}
+
+resource "coder_script" "windows-rdp" {
+  agent_id     = var.agent_id
+  display_name = "windows-rdp"
+  icon         = "https://svgur.com/i/158F.svg" # TODO: add to Coder icons
+  script = <<EOF
+  function Set-AdminPassword {
```

Comment:
suggestion, non-blocking: extract this to a terraform template file

https://developer.hashicorp.com/terraform/language/functions/templatefile